### ISSUE TITLE
crash wifi 8189es buld openwrt 21.02.3 and 21.02.0

### ISSUE THREAD

**Issue by Greg798100** [lyh802/rtl8189ES_linux#2]
Everything is perfectly assembled, but when used, wifi remains inactive and I get system crash.
Used for orangepi r1

**Comment by lyh802** [comment]
if you want to build in openwrt, the branch of "openwrt" should be used, copy folder rtl8189es to openwrt/package/kernel, and make it. it works well on nanopi r1s

**Comment by Greg798100** [comment]
I do this but have crash used wifi

**Comment by lyh802** [comment]
can you provide crash log? I can help to troubleshoot

**Comment by lyh802** [comment]
There should be two wifi devices (ap and client) in the default configuration, if works.

or you can comment NOSTDINC_FLAGS+=-DCONFIG_CONCURRENT_MODE in Makefile

then try again.

**Comment by Greg798100** [comment]
I'm tried different things.
after switching on any module, crash.
https://ibb.co/FbSCLmf

**Comment by lyh802** [comment]
Is there any crash log?

do you use the file?
[mac80211.sh](https://github.com/lyh802/rtl8189ES_linux/blob/openwrt/lib/wifi/mac80211.sh)

**Comment by Greg798100** [comment]
> Is there any crash log?

which log? Im enable wifi modules and boot loop crash


> do you use the file?

your readme not heaved mac80211.sh
Im apply this script after boot devices?

**Comment by r1bnc** [comment]
Hello, just tried to compile v21.02.3 and v21.02.1 and after Enabling WiFi AP mode, the UI became unresponsive and executing `ip a` or `ifconfig` via CLI makes the console freeze  aswell.  it freezes. Just like what @Greg798100  mentioned even after reboot. No crash logs.

Which last version of openWRT does this kernel module work? 
Is it possible to make this work on the latest v22.03.3? 
Any help is much appreciated.

**Comment by Greg798100** [comment]
@r1bnc

> Is it possible to make this work on the latest v22.03.3?<

I continue to use openwrt 18.06.9. It works fine for me wifi 8189, but  Im want new kernel....

**Comment by r1bnc** [comment]
> @r1bnc
> 
> > Is it possible to make this work on the latest v22.03.3?<
> 
> I continue to use openwrt 18.06.9. It works fine for me wifi 8189, but Im want new kernel....

Your openwrt [Orangepi-r1-openwrt](https://github.com/Greg798100/Orangepi-r1-openwrt) build works for me (but cannot do AP and Client at the same time).

I will update you if I get new firmware on Opi R1.

**Comment by lyh802** [comment]
> 
mac80211.sh
open OpenWrt project in GitHub, located to openwrt/package/kernel/mac80211/files/lib/wifi/, you should replace the file.
when build the kernel ,don't build any unnessasery dependencies in wlan drivers.
especilly, if old rtl8189es drivers exists

**Comment by Greg798100** [comment]
> openwrt/package/kernel/mac80211/files/lib/wifi/ <

I did this, there is possibility of  release?

@r1bnc
ok, AP work for me to lan client

**Comment by r1bnc** [comment]
> > openwrt/package/kernel/mac80211/files/lib/wifi/ <
> 
> I did this, there is possibility of release?
> 
> @r1bnc ok, AP work for me to lan client

@Greg798100 
Yes I can do AP only or Client only. Not both at the same time. 



> > 
> 
> mac80211.sh open OpenWrt project in GitHub, located to openwrt/package/kernel/mac80211/files/lib/wifi/, you should replace the file. when build the kernel ,don't build any unnessasery dependencies in wlan drivers. especilly, if old rtl8189es drivers exists

@lyh802 
We try this but the AP and Client not work (orange pi r1 just freeze and no error log no console error), can you share your openwrt with the rtl8189es source code and build in a new repository? I sent you email last week too.

**Comment by lyh802** [comment]
@r1bnc
this is my github-ci
https://github.com/lyh802/openwrt-ci/blob/master/.github/workflows/nanopi-r1s-h3.yml

**Comment by lyh802** [comment]
maybe these patches are what you need
https://github.com/lyh802/openwrt-ci/blob/master/patch/configs/0002-config-mac80211.patch
https://github.com/lyh802/openwrt-ci/blob/master/patch/drivers/0001-add-rtl8189es-driver.patch
